src/power_grid_model/core/serialization.py

```diff
@@ -80,7 +86,31 @@ def load(self) -> Dataset:
             A tuple containing the deserialized dataset in Power grid model input format and the type of the dataset.
         """
         pgc.deserializer_parse_to_buffer(self._deserializer)
-        return self._dataset.get_data()
+
+        filtered_data = self._dataset.get_data()
+
+        def _filter_attributes(attributes):
+            keys_to_remove = []
+            for attr, array in attributes.items():
+                if not isinstance(array, np.ndarray):
+                    continue
+                if is_nan_or_equivalent(array):
+                    keys_to_remove.append(attr)
+            for key in keys_to_remove:
+                del attributes[key]
+
+        if isinstance(self._data_filter, ComponentAttributeFilterOptions):
+            if self._data_filter is ComponentAttributeFilterOptions.RELEVANT:
+                for _, attributes in filtered_data.items():
+                    _filter_attributes(attributes)
```

Comment:
surprising that sonar cloud doesn't warn here, but you could also use
```suggestion
                for attributes in filtered_data.values():
                    _filter_attributes(attributes)
```
not worth changing now though

Reply:
addressed in the latest commit.